src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/Runners/MonitorRunner.cs

```diff
@@ -240,54 +162,17 @@ public async Task StartAsync(CancellationToken token)
             _outputHelper.WriteLine("User Settings Path: {0}", UserSettingsFilePath);
 
             await _adapter.StartAsync(token);
-
-            using IDisposable _ = token.Register(() => CancelCompletionSources(token));
-
-            // Await ready and exited tasks in case process exits before it is ready.
-            if (_runner.ExitedTask == await Task.WhenAny(_readySource.Task, _runner.ExitedTask))
-            {
-                throw new InvalidOperationException("Process exited before it was ready.");
-            }
-
-            // Await ready task to check if it faulted or cancelled.
-            await _readySource.Task;
         }
 
-        private void CancelCompletionSources(CancellationToken token)
+        public virtual async Task WaitForExitAsync(CancellationToken token)
         {
-            _defaultAddressSource.TrySetCanceled(token);
-            _metricsAddressSource.TrySetCanceled(token);
-            _readySource.TrySetCanceled(token);
+            await RunnerExitedTask.WithCancellation(token).ConfigureAwait(false);
+            token.ThrowIfCancellationRequested();
```

Comment:
With the suggested change in WithCancellation, this won't be necessary. It also doesn't really hurt.